app/src/androidTest/java/com/github/factotum_sdp/factotum/MainActivityTest.kt

```diff
@@ -210,11 +210,20 @@ class MainActivityTest {
     @Test
     fun pressingBackOnRBFragmentLeadsOutOfTheApp() {
         LoginFragmentTest.fillUserEntryAndGoToRBFragment()
-        Thread.sleep(3000)
+        Thread.sleep(LOGIN_REFRESH_TIME)
         Espresso.pressBackUnconditionally()
         val uiDevice = UiDevice.getInstance(getInstrumentation())
         assertFalse(uiDevice.currentPackageName == "com.github.factotum_sdp.factotum")
         assertTrue(uiDevice.isScreenOn)
     }
 
+    @Test
+    fun navHeaderDisplaysUserData() {
```

Comment:
If possible, it might be better to check the correctness of the nav header text variables directly. That way we are sure this text is in the right place.